src/server/sample_data/routes/install.js

```diff
@@ -0,0 +1,96 @@
+    validate: {
+      params: Joi.object().keys({
+        id: Joi.string().required(),
+      }).required()
+    },
+    handler: async (request, reply) => {
+      const server = request.server;
+      const sampleDataset = server.getSampleDatasets().find(sampleDataset => {
+        return sampleDataset.id === request.params.id;
+      });
+      if (!sampleDataset) {
+        return reply().code(404);
+      }
+
+      const { callWithRequest } = server.plugins.elasticsearch.getCluster('data');
+      const index = createIndexName(server, sampleDataset.id);
+      const insertCmd = {
+        index: {
+          _index: index
+        }
+      };
+
+      try {
+        const createIndexParams = {
+          index: index,
+          body: {
+            settings: {
+              index: {
+                number_of_shards: 1,
+                number_of_replicas: 0
+              }
+            },
+            mappings: {
+              _doc: {
+                properties: sampleDataset.fields
+              }
+            }
+          }
+        };
+        await callWithRequest(request, 'indices.create', createIndexParams);
+      } catch (err) {
+        return reply(`Unable to create sample data index "${index}", error: ${err.message}`).code(err.status);
+      }
+
+      const now = new Date();
+      const currentTimeMarker = new Date(Date.parse(sampleDataset.currentTimeMarker));
+      function updateTimestamps(doc) {
+        sampleDataset.timeFields.forEach(timeFieldName => {
+          if (doc[timeFieldName]) {
+            doc[timeFieldName] = adjustTimestamp(doc[timeFieldName], currentTimeMarker, now, sampleDataset.preserveDayOfWeekTimeOfDay);
+          }
+        });
+        return doc;
+      }
+      const bulkInsert = async (docs) => {
+        const bulk = [];
+        docs.forEach(doc => {
+          bulk.push(insertCmd);
+          bulk.push(updateTimestamps(doc));
+        });
+        const resp = await callWithRequest(request, 'bulk', { body: bulk });
+        if (resp.errors) {
+          server.log(
+            ['warning'],
+            `sample_data install errors while bulk inserting. Elasticsearch response: ${JSON.stringify(resp, null, ' ')}`);
+          return Promise.reject(new Error(`Unable to load sample data into index "${index}", see kibana logs for details`));
+        }
+      };
+      loadData(sampleDataset.dataPath, bulkInsert, async (err, count) => {
+        if (err) {
+          server.log(['warning'], `sample_data install errors while loading data. Error: ${err}`);
+          return reply(err.message).code(500);
+        }
+
+        try {
+          await request.getSavedObjectsClient().bulkCreate(sampleDataset.savedObjects, { overwrite: true });
+        } catch (err) {
+          server.log(['warning'], `sample_data install errors while loading saved objects. Error: ${err.message}`);
+          return reply(`Unable to load kibana saved objects, see kibana logs for details`).code(403);
```

Comment:
This isn't getting caught here.  I think `bulkCreate` returns an error perhaps instead of throwing an error.  If I have a user who has write access to the sample data indices but read only on .kibana, it shows up as a success but the `launch` button fails to find the dashboard.

Reply:
Looks like that is the case. saved objects are written using a bulk request so an exeception is not thrown and this will have to check for errors.

```
const { items } = await this._writeToCluster('bulk', {
      index: this._index,
      refresh: 'wait_for',
      body: objects.reduce((acc, object) => ([
        ...acc,
        ...objectToBulkRequest(object)
      ]), []),
    });
```